### PR TITLE
Ensure all query keys + B64 variants are encoded.

### PR DESCRIPTION
**Pull request by paulstraw**
Hey @frederickfogerty, it'd be great if you could get take a quick peek at this. :smile:

### CODE REVIEW

**Comment by paulstraw** [comment]
Hah, well uh… I guess that's… working. 👀

File: src/support.js [hunk 1]
```diff
@@ -126,15 +127,20 @@ function processImage (src, longOptions) {
...
+  Object.keys(longOptions).forEach(function (key) {
+    var val = longOptions[key]
+
+    if (PARAM_EXPANSION[key]) {
+      key = PARAM_EXPANSION[key]
     }
+
+    key = encodeURIComponent(key)
+
+    if (key.substr(-2) === '64') {
```
Comment (ekosz) [inline src/support.js]: While `encodeURIComponent` is probably a good idea to add, `js-base64` (a 6.6kb dep) and these next few lines seem very user-landy. As in it doesn't seem to be the most general case that most or many users of this library will get use from.

**Comment by frederickfogerty** [comment]
Yeah I'm with @ekosz on this one, I'm not sure of the motive for this.. don't browsers do this by default anyway?

**Comment by paulstraw** [comment]
Unfortunately, the built-in `atob` and `btoa` browser methods do not output URL-formatted Base64 (which is easy enough to shim with a `String.prototype.replace`), and more importantly don't support any characters outside Latin-1 (which requires a full reimplementation). js-base64 is the lightest, best tool I've found to do this.

Auto Base64 encoding is something that we're in the process of adding to every imgix library, and should be getting very significant usage starting soon.

File: src/index.js [hunk 2]
```diff
@@ -62,11 +62,11 @@ export default class ReactImgix extends Component {
       height: node.scrollHeight,
       mounted: true
     })
-  }
+  };
```
Comment (frederickfogerty) [inline src/index.js]: Not needed with latest babel :)

**Comment by frederickfogerty** [comment]
You're going to put emojis in URLs? That's pretty exciting.

I guess it's not too bad. Maybe I'd like to see a build without base64 encoding though, maybe something like:

```js
import Imgix from 'react-imgix' // No base64 encoding
import Imgix64 from 'react-imgix/base-64'

import b64 from 'js-base64'
Imgix.setBase64Library(b64) // Allow base64 dependency injection
```

File: src/index.js (hunk 2 above)
Comment (paulstraw) [inline src/index.js]: Hmm, that doesn't seem to be the case in 6.4.5, which is the version react-imgix is currently on. Even on a clean checkout, `npm run test` gives:

```
SyntaxError: /Users/paulstraw/dev/imgix/react-imgix/src/index.js: A semicolon is required after a class property (65:3)
  63 |       mounted: true
  64 |     })
> 65 |   }
```

**Comment by paulstraw** [comment]
I'm pretty hesitant to start splitting things up like that. Since js-base64 is only ~6KB unminified/gzipped (and ~1.5KB minified+gzipped). I'm happy to pay that price in order to make sure that all consumers get a consistently good experience, especially given that we're actively promoting this feature with our user base. You cool if I merge this, @frederickfogerty? Happy to reevaluate later if it causes issues.

**Comment by frederickfogerty** [comment]
Yup that sounds fine. Webpack/Rollup will dedupe if the consumer uses it anyway.

File: src/index.js (hunk 2 above)
Comment (frederickfogerty) [inline src/index.js]: Might need to do the old upgrade... `^6.4.5` should match `^6.5` though, which is what I'm on personally

**Pull request merged**